src/compiler/checker.ts

```diff
@@ -45830,19 +45830,20 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
      * the `[Symbol.asyncIterator]()` method first, and then the `[Symbol.iterator]()` method.
      */
     function getIterationTypesOfIterable(type: Type, use: IterationUse, errorNode: Node | undefined) {
-        if (type === silentNeverType) {
+        const reducedType = getReducedType(type);

```

Comment:
Why not just `type = getReducedType(type)`?

Reply:
Done: [`a0a3bb4` (#62661)](https://github.com/microsoft/TypeScript/pull/62661/commits/a0a3bb44121eb932deb190588e607d3312547531)